swift-sdk/Internal/RequestCreator.swift

```diff
@@ -0,0 +1,317 @@
+                AnyHashable.ITBL_KEY_USER: apiUserDict,
+                AnyHashable.ITBL_KEY_ITEMS: itemsToSerialize,
+                AnyHashable.ITBL_KEY_TOTAL: total,
+            ]
+        }
+        
+        return .success(.post(createPostRequest(path: .ITBL_PATH_COMMERCE_TRACK_PURCHASE, body: body)))
+    }
+    
+    func createTrackPushOpenRequest(_ campaignId: NSNumber, templateId: NSNumber?, messageId: String?, appAlreadyRunning: Bool, dataFields: [AnyHashable : Any]?) -> Result<IterableRequest, IterableError> {
+        var body = [AnyHashable : Any]()
+        
+        var reqDataFields: [AnyHashable : Any]
+        if let dataFields = dataFields {
+            reqDataFields = dataFields
+        } else {
+            reqDataFields = [:]
+        }
+        reqDataFields["appAlreadyRunning"] = appAlreadyRunning
+        body[.ITBL_KEY_DATA_FIELDS] = reqDataFields
+        
+        addEmailOrUserId(dict: &body, mustExist: false)
+        
+        body[.ITBL_KEY_CAMPAIGN_ID] = campaignId
+        if let templateId = templateId {
+            body[.ITBL_KEY_TEMPLATE_ID] = templateId
+        }
+        if let messageId = messageId {
+            body[.ITBL_KEY_MESSAGE_ID] = messageId
+        }
+        
+        return .success(.post(createPostRequest(path: .ITBL_PATH_TRACK_PUSH_OPEN, body: body)))
+    }
+    
+    func createTrackEventRequest(_ eventName: String, dataFields: [AnyHashable : Any]?) -> Result<IterableRequest, IterableError> {
+        guard auth.email != nil || auth.userId != nil else {
+            ITBError("Both email and userId are nil")
+            return .failure(IterableError.general(description: "Both email and userId are nil"))
+        }
+        
+        var body = [AnyHashable : Any]()
+        addEmailOrUserId(dict: &body)
+        body[.ITBL_KEY_EVENT_NAME] = eventName
+        if let dataFields = dataFields {
+            body[.ITBL_KEY_DATA_FIELDS] = dataFields
+        }
+        
+        return .success(.post(createPostRequest(path: .ITBL_PATH_TRACK, body: body)))
+    }
+    
+    func createUpdateSubscriptionsRequest(_ emailListIds: [String]?, unsubscribedChannelIds: [String]?, unsubscribedMessageTypeIds: [String]?) -> Result<IterableRequest, IterableError> {
+        var body = [AnyHashable : Any]()
+        addEmailOrUserId(dict: &body)
+        
+        if let emailListIds = emailListIds {
+            body[.ITBL_KEY_EMAIL_LIST_IDS] = emailListIds
+        }
+        if let unsubscribedChannelIds = unsubscribedChannelIds {
+            body[.ITBL_KEY_UNSUB_CHANNEL] = unsubscribedChannelIds
+        }
+        if let unsubscribedMessageTypeIds = unsubscribedMessageTypeIds {
+            body[.ITBL_KEY_UNSUB_MESSAGE] = unsubscribedMessageTypeIds
+        }
+        
+        return .success(.post(createPostRequest(path: .ITBL_PATH_UPDATE_SUBSCRIPTIONS, body: body)))
+    }
+    
+    func createGetInppMessagesRequest(_ count: NSNumber) -> Result<IterableRequest, IterableError> {
```

Comment:
```suggestion
    func createGetInAppMessagesRequest(_ count: NSNumber) -> Result<IterableRequest, IterableError> {
```

"A" missing, maybe all the "app"s should be capitalized as well for consistency